### ISSUE TITLE
SharedObject not working with -Dnext

### ISSUE THREAD

**Issue by Gama11** [openfl/openfl#593]
```haxe
var sharedObject = SharedObject.getLocal("save");
trace(sharedObject.data.value);
sharedObject.data.value = 1;
trace(sharedObject.flush());
```

With `-Dlegacy`, this works as expected:

```
// first run
null
FLUSHED

// second run
1
FLUSHED
```

However, with `-Dnext`,  `value` doesn't seem to be saved (although it claims that it's "FLUSHED") so it traces `null` both times.

Tested on neko with Haxe 3.1.3, OpenFL 3.0.1 and Lime 2.3.1.

**Comment by danim1130** [comment]
Seems like the whole SharedObject class currently only works for js. For example, all the flush() function does is return Flushed (if compiled for anything other than js).

**Comment by larsiusprime** [comment]
Would it work to implement this as a simple JSON text file stored as "openfl_shared_object.json" in the %APPDATA% directory?

**Comment by tarwin** [comment]
I'm guessing we're talking about local only SharedObject? Also, does it need to share both between different instances (web page) and "local" calls from the same instance? Is it restricted per domain?

**Comment by MSGhero** [comment]
Flash SOs are restricted per domain, but any page in that domain can access that SO. They look like they are AMF3 encoded.

\*Feature request\* Async shared object saving for platforms that can support it.

`AppData\Roaming\Macromedia\Flash Player\#SharedObjects\FP8N25XG\uploads.ungrounded.net\566000\566417_main.swf` is where EBF33save.sol is located, I guess from Epic Battle Fantasy 3 on Newgrounds.

**Comment by MSGhero** [comment]
Should this be closed now? 322a036925911387058e1bf9ed8ae1d8685de08f

**Comment by jgranick** [comment]
Yep :smile:
